script.js

```diff
@@ -1,4 +1,175 @@
+    //validation
+    if (!(input === "1" || input === "2")) {
+      return "please input 1 or 2 to decide the order of the dice.";
+    }
+
+    var playerInputNumber = getPlayerNumber(input, player1Dice);
+    player1CombinedNo = playerInputNumber;
+
+    player1Score += playerInputNumber;
+
+    playerNo = 2;
+    rolledDice = false;
+
+    return (
+      "Player 1 combined number is " +
+      player1CombinedNo +
+      ". It is Player 2 turn now."
+    );
+  }
+
+  if (playerNo == 2 && !rolledDice) {
+    return playerRollDice(playerNo, player2Dice);
+  }
+  //validation
+  if (playerNo == 2 && rolledDice) {
+    if (!(input === "1" || input === "2")) {
+      return "please input 1 or 2 to decide the order of the dice.";
+    }
+    var playerInputNumber = getPlayerNumber(input, player2Dice);
+    player2CombinedNo = playerInputNumber;
+    var winningOutput = winningCondition();
+    var reverseWinningOutput = reverseWinningCondition();
+
+    player2Score += playerInputNumber;
+    var playerLeader = leaderboard();
+
+    //reset
+    rolledDice = false;
+    playerNo = 1;
+
+    if (gameMode == "reverse") {
+      isGameModeSet = true;
+      outputValue =
+        "Player 1 combined number is " +
+        player1CombinedNo +
+        ". Player 2 combined number is " +
+        player2CombinedNo +
+        "." +
+        reverseWinningOutput;
+      return outputValue;
+    } else {
+      if (gameMode == "normal") {
+        isGameModeSet = false;
+
+        outputValue =
+          "Player 1 combined number is " +
+          player1CombinedNo +
+          ". Player 2 combined number is " +
+          player2CombinedNo +
+          "." +
+          winningOutput +
+          "<br>" +
+          playerLeader;
+        return outputValue;
+      }
+    }
+  }
+  //leaderboard condition
+  function leaderboard() {
+    if (player1Score > player2Score) {
+      outputMessage =
+        "player 1 is leading the game with score of : " +
+        player1Score +
+        " & player 2 score of : " +
+        player2Score;
+      return outputMessage;
+    } else if (player2Score > player1Score) {
+      outputMessage =
+        "player 2 is leading the game with score of : " +
+        player2Score +
+        " & player 1 score of : " +
+        player1Score;
+      return outputMessage;
+    }
+  }
+};
+
+//wining condition
+function winningCondition() {
+  if (player1CombinedNo > player2CombinedNo) {
+    return " Player 1 Win.";
+  } else if (player1CombinedNo < player2CombinedNo) {
+    return " Player 2 Win.";
+  } else if ((player1CombinedNo = player2CombinedNo)) {
+    return " Its a draw.";
+  }
+}
+//reverse winning cond
+function reverseWinningCondition() {
+  if (player1CombinedNo > player2CombinedNo) {
+    return " Player 2 Win.";
+  } else if (player1CombinedNo < player2CombinedNo) {
+    return " Player 1 Win.";
+  } else if ((player1CombinedNo = player2CombinedNo)) {
+    return " Its a draw.";
+  }
+}
```

Comment:
FYI: If all the if's and else if's are not true in `reverseWinningCondition(),` the code will just return `undefined` if you do not specify a return at the end of the function.